src/main/java/org/ice4j/socket/MultiplexingXXXSocketSupport.java

```diff
@@ -648,9 +648,11 @@ void receive(List<DatagramPacket> received, DatagramPacket p, int timeout)
                 {
                     receivePacket = new DatagramPacket(
                         new byte[POOLED_DATAGRAM_PACKET_SIZE],
-                        0, POOLED_DATAGRAM_PACKET_SIZE);
+                        POOLED_DATAGRAM_PACKET_SIZE);
                 }
 
+                receivePacket.setLength(POOLED_DATAGRAM_PACKET_SIZE);
```

Comment:
I wonder if that's really necessary. I've been reading this code and it looks like there's some confusion:
https://github.com/jitsi/ice4j/blob/master/src/main/java/org/ice4j/socket/DelegatingDatagramSocket.java#L682
How about we reseting both length and offset ?

Reply:
@paweldomas I though it is mandatory, because the [documentation says](https://docs.oracle.com/javase/8/docs/api/java/net/DatagramPacket.html) this
> public void setLength(int length)
Set the length for this packet. The length of the packet is the number of bytes from the packet's data buffer that will be sent, or the number of bytes of the packet's data buffer that will be used for receiving data. The length must be lesser or equal to the offset plus the length of the packet's buffer.

But [it is occurred](https://github.com/jitsi/ice4j/pull/168/files#diff-101ada981bdef99fff62750208d24741R89) that documentation is slightly misleading. There is only one way (I was found it from test) to limit the size of received datagram - is to specify the buffer length [in constructor](https://docs.oracle.com/javase/8/docs/api/java/net/DatagramPacket.html#DatagramPacket-byte:A-int-int-) or via [`void setData(byte[] buf, int offset, int length)`](https://docs.oracle.com/javase/8/docs/api/java/net/DatagramPacket.html#setData-byte:A-int-int-). Changing length via `setLength` before call to `receive` will not limit the size of received data, but if `DatagramPacket` is used for sending it will affect the number of bytes which are sent.
I've added [a unit test](https://github.com/jitsi/ice4j/blob/d160de03848f4d4a2c0617a8d7f53a1f2ff8894d/src/test/java/org/ice4j/util/DatagramPacketBufferTest.java) to verify expected behavior, so further code readers can easily play with sockets and check behavior themselves.

> How about reseting both length and offset ?

There is no way to reset `offset` (no `setOffset` method). 
As I understand `offset` can not be changed neither by `receive` nor by `send`. I believe the reason why datagram packet have such strange signature with `buffer`, `offset` and `length` is to allow programmer to create several `DatagramPacket` instances, which all mapped to single underlying byte array, but with non-overlapping regions specified by `offset` & `length`.


Summarizing, according to [the unit test](https://github.com/jitsi/ice4j/blob/d160de03848f4d4a2c0617a8d7f53a1f2ff8894d/src/test/java/org/ice4j/util/DatagramPacketBufferTest.java) the `setLength` call I've added is not necessary (at least when I run on my machine), but that might be an implementation details. If you want I can remove it.

Reply:
I see, thanks

Reply:
The unit test you've added is nice, but is more like a local experiment as it doesn't test the ice4j itself so I don't think it should land in master.

Reply:
Can you remove the extra commits related to the unit test and force push ?